File: php/models/Usuario.php
<?php 
class Usuario {
    private $nome;
    private $email;
    private $senha;
    private $listas;

    public function __construct($email, $nome, $senha){
        $this->email = $email;
        $this->nome = $nome;
        $this->senha = $senha;
        $this->listas = array();
    }

    public function getNome() {
        return $this->nome;
    }

    public function getSenha() {
        return $this->senha;
    }

    public function getListas() {
        return $this->listas;
    }

    public function adicionarLista($lista) {
        $this->listas[] = $lista;
    }

    private function getIndiceLista($titulo) {
        foreach ($this->listas as $indice => $lista) {
            if ($lista->getTitulo() === $titulo) {
                return $indice;
            }
        }
        return false; // retorna false se a lista não for encontrada
    }

    public function excluirLista($lista) {
        $indice = $this->getIndiceLista($lista->getTitulo()); // encontra o índice da lista no array de listas do usuário
        if ($indice !== false) {
            array_splice($this->listas, $indice, 1); // se o índice for encontrado, remove apenas a lista do indice
        }
    }
}
?>

File: java desktop/src/main/java/com/taniele/java_desktop/model/Usuario.java
package com.taniele.java_desktop.model;

import java.util.ArrayList;
import java.util.List;

public class Usuario {
    private String nome;
    private final String email;
    private String senha;
    private List<Lista> listas;

    public Usuario(String nome, String email, String senha) {
        this.nome = nome;
        this.email = email;
        this.senha = senha;
        this.listas = new ArrayList<>();
    }

    public String getNome() {
        return nome;
    }

    public String getSenha() {
        return senha;
    }

    public List<Lista> getListas() {
        return listas;
    }

    public void adicionarLista(Lista lista){
        listas.add(lista);
    }

    private int getIndice(String titulo) {
        for (int indice = 0; indice < listas.size(); indice++) {
            // Verifica se o titulo da lista armazenada em cada item é igual ao buscado
            if (listas.get(indice).getTitulo().equals(titulo)) {
                return indice;
            }
        }
        return -1;
    }
    public void excluirLista(@org.jetbrains.annotations.NotNull Lista lista){
        int indice = this.getIndice(lista.getTitulo());
        if (indice != -1){
            listas.remove(indice);
        }
    }
}
